pkg/executor/service.go

```diff
@@ -62,8 +63,14 @@ func CreateHeadlessService(
 			},
 		},
 	}
+
+	namingScheme := *cr.Spec.NamingScheme
 	if cr.Status.ClusterService == "" {
-		service.ObjectMeta.GenerateName = headlessSvcNamePrefix
+		if namingScheme == v1beta1.CrNameRole {
+			service.ObjectMeta.Name = cr.Name
```

Comment:
This isn't guaranteed to work. A kdcluster CR name can be of the form `[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)` while a service name must be of the form `[a-z]([-a-z0-9]*[a-z0-9])?`

Reply:
(I think the difference there is just the dots.)

Reply:
If the solution e.g. is just to remove the dots, then we might have to go back to using GenerateName here to guarantee uniqueness.

Reply:
Hmm there is a length different too. A kdcluster name can be 253 chars while a service name only 63.

We could just pass this through the name munge function. It would be a bit conservative in what it truncates for length, since there is no role ID involved, but that's probably fine.